dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -64,40 +70,63 @@ case class Table(
   def put(hashPK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, attributes: _*)
   def put(hashPK: Any, rangePK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, rangePK, attributes: _*)
 
-  def putItem[E <: AnyRef](entity: E)(implicit dynamoDB: DynamoDB): Unit = {
-    val attrs = getAttrValuesToMap(entity, true)
-
-    if (attrs("keys").exists(x => x._1 == "hashPK") && attrs("keys").exists(x => x._1 == "rangePK")) {
-      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
-      val rangePK = attrs("keys").find(f => f._1 == "rangePK").get._2
-      dynamoDB.put(this, hashPK, rangePK, attrs("attributes"): _*)
-    } else if (attrs("keys").exists(x => x._1 == "hashPK")) {
-      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
-      dynamoDB.put(this, hashPK, attrs("attributes"): _*)
+  def putItem[T: u.TypeTag](entity: T)(implicit dynamoDB: DynamoDB): Unit = {
+    val annotations: List[AnnotationMeta] = getterAnnotationsFromEntity(entity)
+    val fields: List[(String, AnyRef)] = getterNamesFromEntity(entity)
+
+    val faJoined = annotations.map(a => {
```

Comment:
can you have an explicit type for `faJoined`? also, the name is not clear enough.

Reply:
changed to annotationFields